tests/clientlib_test.py

```diff
@@ -166,6 +166,28 @@ def test_validate_warn_on_unknown_keys_at_top_level(tmpdir, caplog):
     ]
 
 
+def test_validate_optional_sensible_regex(tmpdir, caplog):
+    f = tmpdir.join('cfg.yaml')
+    f.write(
+        'repos:\n'
+        '-   repo: https://gitlab.com/pycqa/flake8\n'
+        '    rev: 3.7.7\n'
+        '    hooks:\n'
+        '    -   id: flake8\n'
+        '        files: dir/*.py\n',
+    )
+    ret_val = validate_config_main((f.strpath,))
```

Comment:
oh one other thing -- you can call the validate function directly `cfgv.validate({...}, CONFIG_HOOK_DICT)` without needing to write a file out

Reply:
thanks for the note! just out of interest: is there any particular reason why e.g. `test_validate_warn_on_unknown_keys_at_{top|repo}_level` are writing to a temp file?

Reply:
most likely they were older tests / I didn't notice them when reviewing the patches